src/include/processor/operator/persistent/reader/csv/csv_error.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <set>
+#include <string>
+#include <vector>
+
+#include "common/uniq_lock.h"
+
+namespace kuzu::processor {
+
+class BaseCSVReader;
+
+struct LineContext {
+    uint64_t startByteOffset;
+    uint64_t endByteOffset;
+
+    bool isCompleteLine;
+
+    void setNewLine(uint64_t start);
+    void setEndOfLine(uint64_t end);
+};
+
+// If parsing in parallel during parsing we may not be able to determine line numbers
+// Thus we have additional fields that can be used to determine line numbers + reconstruct lines
+// After parsing this will be used to populate a PopulatedCSVError instance
+struct CSVError {
+    std::string message;
+    std::string filePath;
+    LineContext errorLine;
+
+    uint64_t blockIdx;
+    uint64_t numRowsReadInBlock;
+
+    bool operator<(const CSVError& o) const;
+};
+
+struct PopulatedCSVError {
+    std::string message;
+    std::string filePath;
+    std::string reconstructedLine;
```

Comment:
Both PopulatedCSVEerror and CSVError have common fields: message. filePath.
let's have a base class for them.

Reply:
I'd prefer not to do this since I feel like inheritance is overkill in this case (and we won't use the polymorphism). Let me know if you feel strongly about this, I can change it if you do.